overlord/storecontext/context.go

```diff
@@ -247,6 +257,22 @@ func (sc *storeContext) ProxyStoreParams(defaultURL *url.URL) (proxyStoreID stri
 	return "", defaultURL, nil
 }
 
+func (sc *storeContext) StoreAccess() (string, error) {
+	sc.state.Lock()
+	defer sc.state.Unlock()
+
+	access, err := sc.storeAccessQuerier.StoreAccess()
+	if err != nil && !errors.Is(err, state.ErrNoState) {
```

Comment:
What is the difference between the usage of `config.NoOptionError` and `state.ErrNoState`?

Reply:
the former is only used by APIs directly related to the configstate, When we abstract away from those we tend not to use it